importers/link_16_importer.py

```diff
@@ -0,0 +1,199 @@
+import os
+from datetime import datetime, timedelta
+
+from tqdm import tqdm
+
+from pepys_import.core.formats import unit_registry
+from pepys_import.core.formats.location import Location
+from pepys_import.core.validators import constants
+from pepys_import.file.highlighter.support.combine import combine_tokens
+from pepys_import.file.importer import CANCEL_IMPORT, Importer
+from pepys_import.utils.sqlalchemy_utils import get_lowest_privacy
+from pepys_import.utils.text_formatting_utils import (
+    custom_print_formatted_text,
+    format_error_message,
+)
+from pepys_import.utils.unit_utils import convert_absolute_angle, convert_distance, convert_speed
+
+V1_HEADER = "PPLI"
+V2_HEADER = "Xmt/Rcv"
+
+
+class Link16Importer(Importer):
+    """Importer to handle two different formats of track information that are
+    transmitted using Link-16 encoding
+    """
+
+    def __init__(self):
+        super().__init__(
+            name="Link-16 Format Importer",
+            validation_level=constants.BASIC_LEVEL,
+            short_name="Link-16 Importer",
+            default_privacy="Private",
+            datafile_type="Link-16",
+        )
+        self.version = 1
+
+    def can_load_this_type(self, suffix):
+        return suffix.upper() == ".CSV"
+
+    def can_load_this_filename(self, filename):
+        return True
+
+    def can_load_this_header(self, header):
+        # V1 starts w/ PPLI
+        # V2 starts w/ Xmt/Rcv
+        return header.startswith(V1_HEADER) or header.startswith(V2_HEADER)
+
+    def can_load_this_file(self, file_contents):
+        return True
+
+    def _load_this_file(self, data_store, path, file_object, datafile, change_id):
+        # Read the base timestamp from the filename
+        filename, ext = os.path.splitext(os.path.basename(path))
+        datetime_string = self.extract_timestamp(filename)
+        self.base_timestamp = self.timestamp_to_datetime(datetime_string)
+        self.current_hour = 0
+        self.previous_time = timedelta(hours=0, minutes=0, seconds=0)
```

Comment:
hello @mew-nsc - I recognise the role of `previous_time`, but I can't see what `timedelta` is doing here, with the zeroes for h, m, s.

Reply:
This is setting the initial `previous_time` to a value that won't trigger the `current_hour` change on the first line in the file (which will always be 0hrs)

Reply:
I've added some comments in the code around the time handling that hopefully makes things clearer. Please shout if it still doesn't make sense

Reply:
Ok. I've got it.  The `timedelta` docs say that the default value for all fields is zero - so I guess `timedelta()` would be sufficient.